src/Command/Project/ProjectCreateCommand.php

```diff
@@ -216,6 +273,39 @@ protected function getAvailableRegions($runtime = false)
         return $regions;
     }
 
+    /**
+     * Return the catalog.
+     *
+     * The default list is in the config `service.catalog`. This is
+     * replaced at runtime by an API call.
+     *
+     * @param bool $runtime
+     *
+     * @return array
+     */
+    protected function getDefaultCatalog($runtime = false)
+    {
+        if ($runtime) {
+            $catalog = [];
+            $catalog_items = $this->api()->getClient()->getCatalog()->getData();
+            if (!empty($catalog_items)) {
+                foreach ($catalog_items as $item) {
+                    if (isset($item['info']) && isset($item['template'])) {
+                        $catalog[] = $item['template'];
+                    }
+                }
+            }
+        } else {
+            $catalog = (array) $this->config()->get('service.catalog');
+        }
+
+        if (empty($catalog)) {
+            // Should we throw an error here? Do we want to kill the process?
```

Comment:
What happens if the `getTemplateFields()` receives an empty catalog array? If nothing significant -- I'd remove this condition. If something unexpected -- I'd throw an Exception and bail out.

I don't have a clue what am I writing about, right here, so please be free to ignore me if I make no sense at all.

Reply:
If `$catalog` comes back as empty it will cause a logic exception since it's an `asChoice` field. So, do we want it to fail and skip to the next step and just now allow a template choice or stop the process? I'm not sure what the preference is.

Reply:
Hm, I'd personally fail it. What do you think @pjcdawkins?

Reply:
I'd return `[]` here, and let Symfony throw the LogicException